internal/querynodev2/segments/segment_loader.go

```diff
@@ -206,6 +206,14 @@ func (loader *segmentLoader) Load(ctx context.Context,
 		log.Info("no segment to load")
 		return nil, nil
 	}
+	coll := loader.manager.Collection.Get(collectionID)
+	// filter field schema which need to be loaded
+	for _, info := range segments {
```

Comment:
Do we need to inject the `loadFields` in `querypb.SegmentLoadInfo`? Or else the resource estimation in `requestResource` could be wrong.

Reply:
since current implementation totally removes `cold fields`, which means they will not be loaded/accessed/cached. So request resource will not calculate them as well.